packages/devtools_app/lib/src/memory/flutter/memory_heatmap.dart

```diff
@@ -110,10 +112,41 @@ class FlameChartState extends State<FlameChart> with AutoDisposeMixin {
 
     addAutoDisposeListener(controller.searchNotifier, () {
       setState(() {
-        final searchingValue = controller.search;
-        if (searchingValue.isNotEmpty) {
-          final Node node = findNode(searchingValue);
-          selectNode(node);
+        if (controller.search.isNotEmpty) {
+          if (controller.selectTheSearch) {
+            // Select the node.
+            final Node node = findNode(controller.search);
+            selectNode(node);
+
+            closeAutoCompleteOverlay();
+
+            controller.selectTheSearch = false;
+            controller.search = '';
+          } else {
+            // No exact match, return top matches.
+            var matches = matchNames(controller.search);
+            matches.sort();
+            matches = matches.sublist(0, min(topMatches, matches.length));
```

Comment:
you could combine these three calls:
```
final matches = matchNames(controller.search)
  ..sort()
  ..subList(0, min(topMatches, matches.length));
```

Reply:
Not quite because matches is used for end index, but can use ..sort()

```
        var matches = matchNames(controller.search)..sort();
        // No exact match, return top matches.
        matches = matches.sublist(0, min(topMatchesLimit, matches.length));
```